File: client/src/features/api/jobs.ts
import { apiRequest } from './http'
import type { GenerationJob } from '../types/types'

/** Fetch the current state of a generation job by its ID. */
export async function getJob(jobId: string): Promise<GenerationJob> {
  return apiRequest<GenerationJob>(`/jobs/${jobId}`)
}

/** Return true if the job has reached a final state (completed or failed). */
export function isTerminalJob(job: GenerationJob): boolean {
  if (job.status === 'completed' || job.status === 'failed') {
    return true
  }
  return job.stage === 'completed' || job.stage === 'failed'
}

File: server/app/api/routers/jobs.py
from __future__ import annotations

from fastapi import APIRouter

from app.api.schemas.job import GenerationJobModel
from app.services.job_service import get_job_service

router = APIRouter(prefix="/jobs", tags=["jobs"])


@router.get("/{job_id}", response_model=GenerationJobModel)
def get_job(job_id: str):
    """Fetch the current state of a generation job by its ID."""
    return get_job_service(job_id)
